bin/publish

```diff
@@ -1,15 +1,64 @@
-#!/bin/bash -e
-#
-# Push CyberArk sidecar injector mutating webhook service
-# usage: ./bin/publish
-set -ex
+#!/bin/bash
 
-if [ -z "$1" ]
-then
-    echo "tag not provided";
-    exit 1;
-fi
+set -e
 
-cd $(dirname $0)/..
+cd "$(dirname "${0}")"
+. ./build_utils
+cd ..
 
-docker push cyberark/sidecar-injector:$1
+readonly IMAGE_NAME="cyberark/sidecar-injector"
+readonly IMAGE_WITH_TAG="${IMAGE_NAME}:$(git_tag)"
+
+function publish_image() {
+  local tag=${1}
+  image_with_version_tag="${IMAGE_NAME}:${tag}"
+  echo "Tagging and pushing ${image_with_version_tag}"
+
+  docker tag "${IMAGE_WITH_TAG}" "${image_with_version_tag}"
+  docker push "${image_with_version_tag}"
+}
+
+function main() {
+  if [[ $# -gt 0 ]]
```

Comment:
probably worth adding a comment here about this logic - it's intended to be used for non-tag-triggered flows, right? whereas the rest of it is for tags only?

Reply:
Yup, that's it. II'll add a comment now